cmd/private-org-sync/main.go

```diff
@@ -461,34 +461,70 @@ func main() {
 		failOnNonexistentDst: o.failOnNonexistentDst,
 	}
 
-	var syncErrors []error
+	var errs []error
+	locations := make(map[location]struct{})
+
+	if whitelistErrors := getWhitelistedLocations(o.WhitelistOptions.WhitelistConfig.Whitelist, syncer.git, locations); len(whitelistErrors) > 0 {
```

Comment:
Please no side effects. `locations` should be returned as an output, not received as an input and modified.

Reply:
I would prefer to fill the map directly than merging two different maps, which in golang this is really ugly :/

Reply:
I don't understand. If you do

`locations, whitelistErrors := getWhitelistedLocations(whitelist, git)` 

then the rest of this method does need to change at all, and we will not have any confusing side effects. But also, why is merging two maps ugly? It's just an easy iteration.

Reply:
Yea since the whole thing is not complicated we can use the same map later in the code as well. Addressed